### PR TITLE
python3Packages.pyside6-essentials: init at 6.7.0

### PR DESCRIPTION
**Pull request by GetPsyched**
## Description of changes

Adds `pyside6-essentials`. This includes some binaries that are not present in the `pyside6` Python module such as `pyside6-uic`, `pyside6-lrelease`, etc.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by GetPsyched** [comment]
This PR is waiting on #265987. The build fails currently (due to a missing library) but once that PR is merged/this PR is rebased on top of that one, this will build.

**Review by chayleaf — CHANGES_REQUESTED** [review]
this should be built from source instead. I have a suspicion it's already being built by the pyside6 derivation, just not getting installed properly (and I suppose it should be installed in its own output if so)

**Comment by GetPsyched** [comment]
> this should be built from source instead.

Ideally, yeah.

> I have a suspicion it's already being built by the pyside6 derivation, just not getting installed properly

I agree. I was trying to find the binaries I mentioned in the original post in the `pyside6` derivation but I couldn't. I tried some other libraries including `pyside6-addons` and `pyside6-utils` but they didn't have these binaries either; except, `pyside6-essentials` did.

As a side note, I tried to build `pyside6` itself from PyPi but even that didn't include these binaries. Does it mean that the from-source `pyside6` derivation is not bugged and it's just an issue from upstream?

**Comment by chayleaf** [comment]
I don't have the time to look into this atm, but see https://github.com/pyside/pyside-setup/blob/10941bc37450f2fe99cd3afa4e2331df0439d162/create_wheels.py#L196

**Comment by GetPsyched** [comment]
Interesting, this should definitely be in the `pyside6` package itself

**Comment by GetPsyched** [comment]
Now that #265987 is merged, what do we do with this PR?

Given the above discussion, it has been made clear that, ideally, the features from `pyside6-essentials`, `pyside6-addons`, and `pyside6-utils` should come packaged with the `pyside6` package itself.

Calling on the maintainers of `pyside6` to perhaps comment on this. @gebner @Enzime
cc to anyone who knows more about this issue.

**Comment by andersrh** [comment]
I managed to successfully build package and use it on my machine.
It'd be great if it would get merged.

I really needed this for a Python-QT project. Thank you for making this available.

**Comment by GetPsyched** [comment]
Closing this PR since ideally this shouldn't be merged as the wheels for this exist in the `pyside6` module but the Nix package for it doesn't expose them yet. I've already filed an issue for this, track it at #277849.

**Pull request closed without merging**
